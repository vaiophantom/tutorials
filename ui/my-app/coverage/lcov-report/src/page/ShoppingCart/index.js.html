
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/page/ShoppingCart/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/page/ShoppingCart</a> index.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>16/40</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">20% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>3/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">43.24% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>16/37</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { useEffect, useState } from "react";
import ProductCart from "../../components/ProductCart";
import { Button, Form } from "react-bootstrap";
import Header from "../../components/Header";
import { useNavigate } from "react-router-dom";
import { currencyFormat } from "../../utils/helper";
&nbsp;
const ShoppingCart = () =&gt; {
  const [listProducts, setListProducts] = useState([]);
  const [checked, setChecked] = useState(false);
  const [total_payment, setTotal_payment] = useState(0);
  const navigate = useNavigate();
&nbsp;
  useEffect(() =&gt; {
    let listCart = JSON.parse(localStorage.getItem("cart"));
    <span class="missing-if-branch" title="if path not taken" >I</span>if (listCart) {
<span class="cstat-no" title="statement not covered" >      listCart = listCart.map(<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        return { checked: false, ...item };</span>
      });
<span class="cstat-no" title="statement not covered" >      setListProducts(listCart);</span>
    }
  }, []);
&nbsp;
  useEffect(() =&gt; {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (listProducts.length &gt; 0) {
      const hasAllSelected = <span class="cstat-no" title="statement not covered" >listProducts.some(</span>
<span class="fstat-no" title="function not covered" >        (i</span>tem) =&gt; <span class="cstat-no" title="statement not covered" >item.checked === false</span>
      );
<span class="cstat-no" title="statement not covered" >      setChecked(!hasAllSelected);</span>
      const total = <span class="cstat-no" title="statement not covered" >listProducts.reduce(<span class="fstat-no" title="function not covered" >(p</span>rev, current) =&gt; <span class="cstat-no" title="statement not covered" >prev + (current.price*current.quantity),</span> 0);</span>
<span class="cstat-no" title="statement not covered" >      setTotal_payment(total)</span>
    }
  }, [listProducts]);
&nbsp;
  const onSelectAll = <span class="fstat-no" title="function not covered" >(e</span>) =&gt; {
<span class="cstat-no" title="statement not covered" >    setChecked(e.target.checked);</span>
    const newListProducts = <span class="cstat-no" title="statement not covered" >listProducts.map(<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      return { ...item, checked: e.target.checked };</span>
    });
<span class="cstat-no" title="statement not covered" >    setListProducts(newListProducts);</span>
  };
&nbsp;
  const outputCheckBox = <span class="fstat-no" title="function not covered" >(d</span>ata) =&gt; {
<span class="cstat-no" title="statement not covered" >    if (data?.id) {</span>
      const index = <span class="cstat-no" title="statement not covered" >listProducts.findIndex(<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; <span class="cstat-no" title="statement not covered" >item.id === data.id)</span>;</span>
<span class="cstat-no" title="statement not covered" >      listProducts[index].checked = data.checked;</span>
<span class="cstat-no" title="statement not covered" >      setListProducts([...listProducts]);</span>
    }
  };
&nbsp;
  const deleteItem = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const newList = <span class="cstat-no" title="statement not covered" >listProducts.filter(<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; <span class="cstat-no" title="statement not covered" >item.checked == false)</span>;</span>
<span class="cstat-no" title="statement not covered" >    localStorage.setItem("cart", JSON.stringify(newList));</span>
<span class="cstat-no" title="statement not covered" >    setListProducts(newList);</span>
  };
&nbsp;
  const onPayment = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    navigate("/checkout");</span>
  };
  const handleSearch = <span class="fstat-no" title="function not covered" >(t</span>ype, value) =&gt; {};
&nbsp;
  return (
    &lt;div className="shopping-cart-screen"&gt;
      &lt;Header handleSearch={handleSearch}/&gt;
      &lt;div className="shopping-cart-body"&gt;
        &lt;div className="shopping-cart-label"&gt;
          &lt;div style={{ flex: 3, display: "flex", flexDirection: "row" }}&gt;
            &lt;Form.Check
              checked={checked}
              onChange={onSelectAll}
              style={{ marginRight: 20 }}
              type="checkbox"
              id="select_all"
            &gt;&lt;/Form.Check&gt;
            &lt;p&gt;Product&lt;/p&gt;
          &lt;/div&gt;
          &lt;div style={{ flex: 1 }}&gt;Price&lt;/div&gt;
          &lt;div style={{ flex: 1 }}&gt;Quantity&lt;/div&gt;
          &lt;div style={{ flex: 1 }}&gt;Total Price&lt;/div&gt;
        &lt;/div&gt;
        &lt;div className="shopping-cart-product"&gt;
          {listProducts.map(<span class="fstat-no" title="function not covered" >(i</span>tem) =&gt; {
<span class="cstat-no" title="statement not covered" >            return &lt;ProductCart data={item} outputCheckBox={outputCheckBox} /&gt;;</span>
          })}
        &lt;/div&gt;
      &lt;/div&gt;
      &lt;div className="shopping-cart-footer"&gt;
        &lt;Button onClick={deleteItem} variant="warning" style={{ flex: 1 }}&gt;
          Delete
        &lt;/Button&gt;
        &lt;div style={{ flex: 5, marginRight: 20, textAlign: "end" }}&gt;
          &lt;p style={{fontWeight:"bold"}}&gt;Total Payment: {currencyFormat(total_payment)} $&lt;/p&gt;
        &lt;/div&gt;
        &lt;Button
          disabled={listProducts.length == 0}
          onClick={onPayment}
          variant="warning"
          style={{ flex: 2 }}
        &gt;
          Checkout
        &lt;/Button&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
export default ShoppingCart;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-08-10T10:03:11.925Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    