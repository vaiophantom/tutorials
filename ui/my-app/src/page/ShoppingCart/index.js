import { useEffect, useState } from "react";
import ProductCart from "../../components/ProductCart";
import { Button, Form } from "react-bootstrap";
import Header from "../../components/Header";
import { useNavigate } from "react-router-dom";
import { currencyFormat } from "../../utils/helper";

const ShoppingCart = () => {
  const [listProducts, setListProducts] = useState([]);
  const [checked, setChecked] = useState(false);
  const [total_payment, setTotal_payment] = useState(0);
  const navigate = useNavigate();

  useEffect(() => {
    let listCart = JSON.parse(localStorage.getItem("cart"));
    if (listCart) {
      listCart = listCart.map((item) => {
        return { checked: false, ...item };
      });
      setListProducts(listCart);
    }
  }, []);

  useEffect(() => {
    if (listProducts.length > 0) {
      const hasAllSelected = listProducts.some(
        (item) => item.checked === false
      );
      setChecked(!hasAllSelected);
      const total = listProducts.reduce((prev, current) => prev + (current.price*current.quantity), 0);
      setTotal_payment(total)
    }
  }, [listProducts]);

  const onSelectAll = (e) => {
    setChecked(e.target.checked);
    const newListProducts = listProducts.map((item) => {
      return { ...item, checked: e.target.checked };
    });
    setListProducts(newListProducts);
  };

  const outputCheckBox = (data) => {
    if (data?.id) {
      const index = listProducts.findIndex((item) => item.id === data.id);
      listProducts[index].checked = data.checked;
      setListProducts([...listProducts]);
    }
  };

  const deleteItem = () => {
    const newList = listProducts.filter((item) => item.checked == false);
    localStorage.setItem("cart", JSON.stringify(newList));
    setListProducts(newList);
  };

  const onPayment = () => {
    navigate("/checkout");
  };
  const handleSearch = (type, value) => {};

  return (
    <div className="shopping-cart-screen">
      <Header handleSearch={handleSearch}/>
      <div className="shopping-cart-body">
        <div className="shopping-cart-label">
          <div style={{ flex: 3, display: "flex", flexDirection: "row" }}>
            <Form.Check
              checked={checked}
              onChange={onSelectAll}
              style={{ marginRight: 20 }}
              type="checkbox"
              id="select_all"
            ></Form.Check>
            <p>Product</p>
          </div>
          <div style={{ flex: 1 }}>Price</div>
          <div style={{ flex: 1 }}>Quantity</div>
          <div style={{ flex: 1 }}>Total Price</div>
        </div>
        <div className="shopping-cart-product">
          {listProducts.map((item) => {
            return <ProductCart data={item} outputCheckBox={outputCheckBox} />;
          })}
        </div>
      </div>
      <div className="shopping-cart-footer">
        <Button onClick={deleteItem} variant="warning" style={{ flex: 1 }}>
          Delete
        </Button>
        <div style={{ flex: 5, marginRight: 20, textAlign: "end" }}>
          <p style={{fontWeight:"bold"}}>Total Payment: {currencyFormat(total_payment)} $</p>
        </div>
        <Button
          disabled={listProducts.length == 0}
          onClick={onPayment}
          variant="warning"
          style={{ flex: 2 }}
        >
          Checkout
        </Button>
      </div>
    </div>
  );
};
export default ShoppingCart;
