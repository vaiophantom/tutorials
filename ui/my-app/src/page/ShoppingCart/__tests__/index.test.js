import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ShoppingCart from '../';

it('renders Detail', async () => {
 const {container} = render(<ShoppingCart />, {wrapper: BrowserRouter});
 expect(container).toMatchSnapshot();
});
