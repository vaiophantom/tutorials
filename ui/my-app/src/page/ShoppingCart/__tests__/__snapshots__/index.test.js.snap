// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Detail 1`] = `
<div>
  <div
    class="shopping-cart-screen"
  >
    <div
      class="header"
    >
      <img
        class="logo"
        height="100"
        src="logo.png"
        width="200"
      />
      <div
        class="search"
      >
        <div
          class="input-group"
          style="background-color: white; align-items: center;"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-outline-primary"
            type="button"
          >
            name
          </button>
          <div
            aria-labelledby=""
            class="dropdown-menu"
            x-placement="bottom-start"
          >
            <a
              class="dropdown-item"
              data-rr-ui-dropdown-item=""
              href="#"
              role="button"
              tabindex="0"
            >
              name
            </a>
            <a
              class="dropdown-item"
              data-rr-ui-dropdown-item=""
              href="#"
              role="button"
              tabindex="0"
            >
              branch
            </a>
            <a
              class="dropdown-item"
              data-rr-ui-dropdown-item=""
              href="#"
              role="button"
              tabindex="0"
            >
              category
            </a>
            <a
              class="dropdown-item"
              data-rr-ui-dropdown-item=""
              href="#"
              role="button"
              tabindex="0"
            >
              color
            </a>
          </div>
          <input
            aria-describedby="basic-addon3"
            class="form-control"
            placeholder="Search products based on category name, product name, price, branch..."
            value=""
          />
          <img
            class=""
            height="30"
            src="search.jpeg"
            width="30"
          />
        </div>
      </div>
      <img
        class="cart"
        height="40"
        src="cart.png"
        width="40"
      />
    </div>
    <div
      class="shopping-cart-body"
    >
      <div
        class="shopping-cart-label"
      >
        <div
          style="flex: 3; display: flex; flex-direction: row;"
        >
          <div
            class=""
            style="margin-right: 20px;"
          >
            <input
              class="form-check-input"
              id="select_all"
              type="checkbox"
            />
          </div>
          <p>
            Product
          </p>
        </div>
        <div
          style="flex: 1;"
        >
          Price
        </div>
        <div
          style="flex: 1;"
        >
          Quantity
        </div>
        <div
          style="flex: 1;"
        >
          Total Price
        </div>
      </div>
      <div
        class="shopping-cart-product"
      />
    </div>
    <div
      class="shopping-cart-footer"
    >
      <button
        class="btn btn-warning"
        style="flex: 1;"
        type="button"
      >
        Delete
      </button>
      <div
        style="flex: 5; margin-right: 20px; text-align: end;"
      >
        <p
          style="font-weight: bold;"
        >
          Total Payment: 
          0.00
           $
        </p>
      </div>
      <button
        class="btn btn-warning"
        disabled=""
        style="flex: 2;"
        type="button"
      >
        Checkout
      </button>
    </div>
  </div>
</div>
`;
